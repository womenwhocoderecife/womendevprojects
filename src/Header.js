import React from 'react';

import header from './assets/style/header.css';
import logo from './assets/images/logo.png';

const Header = () => (
  <header className="header">
    <figure className="header__figure">
      <img className="header__img" src={logo} alt="logo" />
      <h1 className="header__text">
        Juntas! Desenvolvendo Tecnologia com Impacto
      </h1>
    </figure>
  </header>
);

export default Header;
