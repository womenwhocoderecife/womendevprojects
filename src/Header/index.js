import React from 'react';

import Menu from '../Menu';

import logo from '../assets/images/logo.png';

import './index.css';

const Header = () => (
  <header className="header">
    <Menu />
    <a className="menu__logo" href="/">
      <figure className="menu__figure header__figure">
        <img
          className="menu__img"
          src={logo}
          alt="Logo Women Dev Projects"
        />
      </figure>
    </a>
    <h1 className="header__text">
      Juntas! Desenvolvendo Tecnologia com Impacto
    </h1>
  </header>
);

export default Header;
