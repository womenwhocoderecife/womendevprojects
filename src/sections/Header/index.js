import React, { useState, useEffect } from 'react';
import logo from '../../assets/images/logo.png';

import {
  HeaderStyle,
  Container,
  LogoLink,
  Logo,
  Nav,
  NavList,
  ListItem,
  ListLink,
  MenuToggle,
  ToggleOne,
  ToggleTwo,
  ToggleThree,
} from './styles';

export default function Header() {
  const [menuVisibility, setMenuVisibility] = useState(false);
  const [activeSection, setActiveSection] = useState('#hero');
  const [scroll, setScroll] = useState(window.scrollY);

  useEffect(() => {
    window.addEventListener('scroll', () => {
      setScroll(window.scrollY);
      const observer = new IntersectionObserver(entries => {
      
        entries.forEach(entry => {
          const id = entry.target.getAttribute('id');

          if (entry.isIntersecting && entry.intersectionRatio > 0.4) {
            setActiveSection(`#${id}`);
          }
        });
      });
      document
        .querySelectorAll('section[id]')
        .forEach(section => observer.observe(section));
    });
  }, []);

  return (
    <HeaderStyle scroll={scroll}>
      <Container>
        <LogoLink
          href="#hero"
        >
          <Logo src={logo} alt="Women Dev Projects" />
        </LogoLink>
        <MenuToggle
          rule="button"
          aria-label="Abrir menu"
          onClick={() => setMenuVisibility(!menuVisibility)}
        >
          <ToggleOne menuVisibility={menuVisibility} />
          <ToggleTwo menuVisibility={menuVisibility} />
          <ToggleThree menuVisibility={menuVisibility} />
        </MenuToggle>
        <Nav menuVisibility={menuVisibility}>
          <NavList menuVisibility={menuVisibility}>
            <ListItem>
              <ListLink
                href="#hero"
                activeSection={activeSection}
                onClick={() => setMenuVisibility(false)}
              >
                Home
              </ListLink>
            </ListItem>
            <ListItem>
              <ListLink
                href="#about"
                activeSection={activeSection}
                onClick={() => setMenuVisibility(false)}
              >
                Sobre
              </ListLink>
            </ListItem>
            <ListItem>
              <ListLink
                href="#proposal"
                activeSection={activeSection}
                onClick={() => setMenuVisibility(false)}
              >
                Nossa Proposta
              </ListLink>
            </ListItem>
            <ListItem>
              <ListLink
                href="#projects"
                activeSection={activeSection}
                onClick={() => setMenuVisibility(false)}
              >
                Projetos
              </ListLink>
            </ListItem>
            <ListItem>
              <ListLink
                href="#team"
                activeSection={activeSection}
                onClick={() => setMenuVisibility(false)}
              >
                Equipe
              </ListLink>
            </ListItem>
            <ListItem>
              <ListLink
                href="#contact"
                activeSection={activeSection}
                onClick={() => setMenuVisibility(false)}
              >
                Contato
              </ListLink>
            </ListItem>
          </NavList>
        </Nav>
      </Container>
    </HeaderStyle>
  );
}