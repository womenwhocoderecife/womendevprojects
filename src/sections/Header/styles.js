import styled from 'styled-components';

export const HeaderStyle = styled.header`
  position: fixed;
  width: 100%;
  height: 144px;
  z-index: 99999999;

  background-color: ${(props) =>
    props.scroll > 0 ? props.theme.colors.violet : 'none'};

  @media (max-width: 507px) {
    height: 80px;
  }
`;

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 90%;
  max-width: 980px;
  padding: 1rem;
  margin: auto;

  @media (max-width: 1024px) {
    width: 100%;
  }
`;

export const LogoLink = styled.a`
  z-index: 99999999;
  @media (max-width: 1024px) {
    width: 16%;
  }

  @media (max-width: 420px) {
    width: 24%;
  }
`;

export const Logo = styled.img`
  width: 80%;
  @media (min-width: 768px) and (max-width: 1024px) {
    width: 62%;
  }
`;

export const Nav = styled.nav`
  width: 100%;

  @media (max-width: 1024px) {
    position: absolute;
    display: ${(props) => (props.menuVisibility ? 'flex' : 'none')};
    justify-content: center;
    width: 2%;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background-color: #965c96;
    z-index: 10;
  }
`;

export const NavList = styled.ul`
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  padding-left: 0;
  list-style: none;

  @media (max-width: 1024px) {
    display: ${(props) => (props.menuVisibility ? 'flex' : 'none')};
    flex-direction: column;
    text-align: center;
  }
`;

export const ListItem = styled.li`
  font-family: ${(props) => props.theme.fonts.join(', ')};
  font-style: normal;
  font-weight: bold;
  font-size: 1.6rem;
  list-style: none;
`;

export const ListLink = styled.a`
  text-decoration: none;
  text-transform: uppercase;
  font-size: 1.2rem;
  cursor: pointer;
  color: ${(props) => props.theme.colors.lightSecondary};
  transition: all 0ms linear 0s;

  :hover {
    color: #cea1ff;
    border-bottom: 2px solid #ff50cd;
  }

  &[href='${(props) => props.activeSection}'] {
    color: #ce9ffe;
    border-bottom: 2px solid #ff50cd;
  }

  @media (max-width: 1024px) {
    font-size: 2rem;
    transition-duration: 0.5s;
  }

  @media (max-width: 420px) {
    font-size: 1.4rem;
  }
`;

export const MenuToggle = styled.div`
  @media (max-width: 1024px) {
    width: 40px;
    height: 30px;
    z-index: 16;
  }
`;

const ToggleLine = styled.div`
  @media (max-width: 1024px) {
    height: 5px;
    width: 100%;
    margin: 6px auto;
    transition-duration: 0.3s;
    background-color: ${(props) => props.theme.colors.lightSecondary};
  }
`;

export const ToggleOne = styled(ToggleLine)`
  transform: ${(props) =>
    props.menuVisibility ? 'rotate(45deg) translate(7px, 7px)' : ''};
`;

export const ToggleTwo = styled(ToggleLine)`
  opacity: ${(props) => (props.menuVisibility ? 0 : 1)};
`;

export const ToggleThree = styled(ToggleLine)`
  transform: ${(props) =>
    props.menuVisibility
      ? 'rotate(-45deg) translate(8px, -9px)'
      : ''};
`;
