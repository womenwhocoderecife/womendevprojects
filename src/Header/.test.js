import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Header from '.';

describe('Header', () => {
  it('renders without crashing', () => {
    const header = document.createElement('header');
    ReactDOM.render(<Header />, header);
    ReactDOM.unmountComponentAtNode(header);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
