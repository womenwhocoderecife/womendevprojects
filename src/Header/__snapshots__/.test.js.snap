// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header has a valid snapshot 1`] = `
<header
  className="header"
>
  <div
    className="menu-container"
  >
    <button
      className="menu__button"
      onClick={[Function]}
    >
      ☰ Menu
    </button>
    <button
      className="menu__button menu--show-menu"
      onClick={[Function]}
    >
      ☰ Menu
    </button>
    <nav
      className="menu menu--show-menu"
    >
      <a
        className="menu__logo"
        href="/"
      >
        <figure
          className="menu__figure"
        >
          <img
            alt="Logo Women Dev Projects"
            className="menu__img"
            src="logo.png"
          />
        </figure>
      </a>
      <ul
        className="menu__list"
      >
        <li
          className="menu__list__item"
        >
          <a
            className="menu__list__link"
            href="#home"
          >
            Home
          </a>
        </li>
        <li
          className="menu__list__item"
        >
          <a
            className="menu__list__link"
            href="#about"
          >
            Sobre
          </a>
        </li>
        <li
          className="menu__list__item"
        >
          <a
            className="menu__list__link"
            href="#proposal"
          >
            Nossa Proposta
          </a>
        </li>
        <li
          className="menu__list__item"
        >
          <a
            className="menu__list__link"
            href="#team"
          >
            Equipe
          </a>
        </li>
        <li
          className="menu__list__item"
        >
          <a
            className="menu__list__link"
            href="#contact"
          >
            Contato
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <a
    className="menu__logo"
    href="/"
  >
    <figure
      className="menu__figure header__figure"
    >
      <img
        alt="Logo Women Dev Projects"
        className="menu__img"
        src="logo.png"
      />
    </figure>
  </a>
  <h1
    className="header__text"
  >
    Juntas! Desenvolvendo Tecnologia com Impacto
  </h1>
</header>
`;
